### PR TITLE
autocomplete job name/id

### PR DESCRIPTION
**Pull request by asvetlov**

### CODE REVIEW

File: neuromation/cli/click_types.py [hunk 1]
```diff
@@ -193,3 +194,41 @@ async def async_complete(
...
+                job_name = job.name or ""
+                for test in (
+                    job.id,
+                    job_name,
+                    f"job:{job.id}",
+                    f"job:/{client.username}/{job.id}",
+                    f"job://{client.cluster_name}/{client.username}/{job.id}",
+                ):
+                    if test.startswith(incomplete):
+                        ret.append((test, job_name))
```
Comment (serhiy-storchaka) [inline neuromation/cli/click_types.py:229]: What is the meaning of this tuple? I have not find the documentation.

Comment (asvetlov) [reply neuromation/cli/click_types.py:229]: The first item is the found match, the second is for help string (used in `zsh` completion only).

File: neuromation/cli/click_types.py [hunk 2]
```diff
@@ -193,3 +194,41 @@ async def async_complete(
...
+            now = datetime.now()
+            async for job in client.jobs.list(
+                since=now - timedelta(days=7), reverse=True, limit=100
+            ):
+                job_name = job.name or ""
+                for test in (
+                    job.id,
+                    job_name,
+                    f"job:{job.id}",
+                    f"job:/{client.username}/{job.id}",
```
Comment (serhiy-storchaka) [inline neuromation/cli/click_types.py:225]: If you use the form `job:/{user-name}/{job-id}`, `user-name` is usually the other user's name. You can use this name for filtering.

If you use the form `job:{job-id}` (without `/` after `job:`), you refer to your own jobs, so you can pass `owners=[client.username]` to `list()`.

File: neuromation/cli/click_types.py (hunk 1 above)
Comment (asvetlov) [reply neuromation/cli/click_types.py:229]: I've found that just job ID doesn't help too much to select the job.
The help text can contain more info, suggestions are welcome!

Comment (serhiy-storchaka) [reply neuromation/cli/click_types.py:229]: Then you should return `job_name` as the first item if `job_name.startswith(incomplete)` and an URI if `incomplete` matches `job:...`.

File: neuromation/cli/click_types.py (hunk 2 above)
Comment (asvetlov) [reply neuromation/cli/click_types.py]: thanks, now I take cluster and user from job-description.

Regarding `owners=[client.username]` -- I prefer to make the single REST call for the whole autocomplete, it makes code much simpler. Do you have an objection?

Comment (serhiy-storchaka) [reply neuromation/cli/click_types.py]: Ok, it works now. Maybe adding additional parameters taken from `incomplete` could make the request faster for admins, but I am not sure the difference will be significant.

But I think that you should return a name or an URI when autocomplete a name or an URI.

Comment (serhiy-storchaka) [reply neuromation/cli/click_types.py]: Please ignore the last phrase. I missed that the current code does exactly this.

**Comment by asvetlov** [comment]
Thanks!

**Pull request merged**